central/complianceoperator/v2/report/manager/format/formatter.go

```diff
@@ -127,26 +130,42 @@ func generateRecord(row *report.ResultRow) []string {
 	}
 }
 
-func (f *FormatterImpl) createFailedClusterFileInZip(zipWriter ZipWriter, filename string, failedCluster *storage.ComplianceOperatorReportSnapshotV2_FailedCluster) error {
+func (f *FormatterImpl) createFailedClusterFileInZip(zipWriter ZipWriter, filename string, failedCluster *report.FailedCluster) error {
 	w, err := zipWriter.Create(filename)
 	if err != nil {
 		return err
 	}
 	csvWriter := f.newCSVWriter(failedClusterCSVHeader, true)
-	csvWriter.AddValue(generateFailRecord(failedCluster))
+	for _, reason := range failedCluster.Reasons {
+		csvWriter.AddValue(generateFailRecord(failedCluster.ClusterId, failedCluster.ClusterName, reason, failedCluster.OperatorVersion))
+	}
 	return csvWriter.WriteCSV(w)
 }
 
-func generateFailRecord(failedCluster *storage.ComplianceOperatorReportSnapshotV2_FailedCluster) []string {
+func generateFailRecord(clusterID, clusterName, reason, coVersion string) []string {
 	// The order in the slice needs to match the order defined in `failedClusterCSVHeader`
 	return []string{
-		failedCluster.GetClusterId(),
-		failedCluster.GetClusterName(),
-		strings.Join(failedCluster.GetReasons(), ", "),
-		failedCluster.GetOperatorVersion(),
+		clusterID,
+		clusterName,
+		reason,
+		coVersion,
 	}
 }
 
+func getProfileDiff(allProfiles []string, failedCluster *report.FailedCluster) []string {
```

Comment:
note: calling this func with duplicates in the slice (and the second param) will remove the duplicates. So diff from (a,a,a) & (a,a,a,a) will be empty.
If that is intended, then all good, if not, then now you know :)